tiers/models.py

```diff
@@ -56,12 +48,17 @@ def __str__(self):
     class Meta:
         app_label = 'tiers'
 
-    @check_if_exempt
-    def has_tier_expired(self):
+    def get_tier_info(self):
+        return TierInfo(
+            tier=self.name,
+            subscription_ends=self.tier_expires_at,
+            always_active=self.tier_enforcement_exempt,
+        )
+
+    def has_tier_expired(self, now=None):
         """Helper function that checks whether a tier has expired"""
-        return timezone.now() > self.tier_expires_at
+        return self.get_tier_info().has_subscription_ended(now=now)
 
-    @check_if_exempt
     def time_til_tier_expires(self, now=None):
```

Comment:
Thank you @OmarIthawi . All is clear except for this method it looks confusing after adding `time_til_expiration` . I believe we should remove it completely. You'll find in the middleware
```python
request.session['TIER_EXPIRES_IN'] = tier_info.time_til_expiration()
```
then a comparison with this method in tests
```python
assert self.request.session['TIER_EXPIRES_IN'] == self.tier.time_til_tier_expires()
```
what do you think?

Reply:
You're right. Tests are stale and needs refactoring.

I'm making safe refactoring for the model until we fully decouple the middleware from the `Tier` model.